# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [ ]:
a = 101
b = 210
c = 111
d = 402
((a + b) * c) / d

85.8731343283582

In [ ]:
a = 101
b = 210
c = 111
d = 402
(a + b) * c / d

85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [ ]:
11283 % 89

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [ ]:
11283 // 89

126

In [ ]:
a = 11283  #це просто граюсь
b = 89
round(a / b, 3 )

126.775

4. Запишіть у змінну `name` ваше імʼя.

In [ ]:
name = 'Hanna'

5. Запишіть у змінну `surname` ваше прізвище.

In [ ]:
surname = 'Ivanova'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [ ]:
name_surname = name + ' ' + surname
name_surname

'Hanna Ivanova'

Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [ ]:
print(name_surname.upper())

HANNA IVANOVA


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

In [ ]:
print(name_surname.lower())

hanna ivanova


8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [ ]:
name_surname[2]

'n'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [ ]:
surname[0], surname[6]

('I', 'a')

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [ ]:
print(len(surname))

7


Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [ ]:
city = 'Antananarivo'
letter_count = (city.lower().count('a'))
letter_count

4

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [ ]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
print(my_string.startswith("The"))
print(my_string.endswith("dog!"))

True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [ ]:
split_words = my_string.split()
split_words

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']

In [ ]:
print("_".join(split_words))

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`